src/Content/Text/BBCode.php

```diff
@@ -1361,6 +1361,12 @@ function ($match) {
 				. '</a>';
 		}, $text);
 
+		// We need no target="_blank" for local links
+		// convert links start with System::baseUrl() as local link without the target="_blank" attribute
+		$escapedBaseUrl = preg_quote(System::baseUrl(), '/');
+		$text = preg_replace("/\[url\]($escapedBaseUrl)([$URLSearchString]*)\[\/url\]/ism", '<a href="$1$2">$1$2</a>', $text);
```

Comment:
It looks like you don't need that many capture groups.
```suggestion
		$text = preg_replace("/\[url\]($escapedBaseUrl[$URLSearchString]*)\[\/url\]/ism", '<a href="$1">$1</a>', $text);
```